script.js

```diff
@@ -44,6 +43,10 @@ window.GaiaAppIcon = (function(exports) {
     this._hasIcon = false;
     this._hasUserSetIcon = false;
     this._hasPredefinedIcon = false;
+    this._iconUrl = null;
+    this._pendingIconUrl = null;
+    this._pendingIconRefresh = false;
+    this._lastState = 'unavailable';
```

Comment:
I don't think 'unavailable' is a valid state, I'd rather it be initialised to null.